src/main/scala/io/iohk/ethereum/blockchain/sync/regular/BlockFetcher.scala

```diff
@@ -198,31 +199,37 @@ class BlockFetcher(
       }
       supervisor ! ProgressProtocol.GotNewBlock(newState.knownTop)
       fetchBlocks(newState)
-    case MessageFromPeer(NewBlock(_, block, _), peerId) =>
-      //TODO ETCM-389: Handle mined, checkpoint and new blocks uniformly
-      log.debug("Received NewBlock {}", block.idTag)
-      val newBlockNr = block.number
-      val nextExpectedBlock = state.lastBlock + 1
-
-      if (state.isOnTop && newBlockNr == nextExpectedBlock) {
-        log.debug("Passing block directly to importer")
-        val newState = state.withPeerForBlocks(peerId, Seq(newBlockNr)).withKnownTopAt(newBlockNr)
-        state.importer ! OnTop
-        state.importer ! ImportNewBlock(block, peerId)
-        supervisor ! ProgressProtocol.GotNewBlock(newState.knownTop)
-        context become started(newState)
-      } else {
-        log.debug("Ignoring received block as it doesn't match local state or fetch side is not on top")
-        val newState = state.withPossibleNewTopAt(block.number)
-        supervisor ! ProgressProtocol.GotNewBlock(newState.knownTop)
-        fetchBlocks(newState)
-      }
+    case MessageFromPeer(CommonMessages.NewBlock(block, _), peerId) =>
```

Comment:
Maybe we could add a comment regarding the needs of adding a pattern match in case new versions of NewBlock messages are added.